### PR TITLE
CORENET-7375: d/s merge Bump Kubernetes to 1.36.2 and Go to 1.26

### PR DESCRIPTION
**Pull request by miheer**
Bump Kubernetes to 1.36.2 and Go to 1.26

## Summary by CodeRabbit

* **Chores**
  * Updated the Go toolchain to 1.26 across the build container and CI (build + test runs).
  * Refreshed Prometheus, Kubernetes, and related dependencies (including module pins).
* **Bug Fixes**
  * Improved webhook TLS certificate change detection to trigger reloads more reliably when certificate files are updated, including atomic-style replacements.
* **Tests**
  * Added coverage for certificate change detection: unchanged files, in-place updates, and atomic-style replacements.

### CODE REVIEW

File: cmd/webhook/main.go [hunk 1]
```diff
@@ -130,34 +126,34 @@ func main() {
...
 		time.Sleep(1 * time.Second)
 	}
 
 }
 
+func certificateFileChanged(previous, current os.FileInfo) bool {
+	return !os.SameFile(previous, current) ||
+		previous.Size() != current.Size() ||
+		!previous.ModTime().Equal(current.ModTime())
+}
```
[the remark this answers is not in the document]
Comment (bpickard22) [reply cmd/webhook/main.go:155]: I think the odds of hitting this in a prod env are so low that we can ignore this, if we ever hit this edge case we can re-eval

**Comment by miheer** [comment]
[ci/prow/e2e-aws](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_multus-admission-controller/123/pull-ci-openshift-multus-admission-controller-main-e2e-aws/2078170624877924352) — Job failed.
Multus successfully attached networking to the failing pods (AddedInterface events), and the admission-controller image built successfully. The 10 network-segmentation failures are informing only and come from OVN-Kubernetes tests, mostly invalid dual-CIDR input on an IPv4 cluster.
So, will re-run again.

**Comment by miheer** [comment]
I have to yet create a upstream PR for the bump for this in [k8snetworkplumbingwg/net-attach-def-admission-controller](https://github.com/k8snetworkplumbingwg/net-attach-def-admission-controller)

**Comment by miheer** [comment]
just created an upstream PR https://github.com/k8snetworkplumbingwg/net-attach-def-admission-controller/pull/78

**Comment by miheer** [comment]
@bpickard22 I think we should consider just a pre-merge testing PR for testing upstream bump https://github.com/k8snetworkplumbingwg/net-attach-def-admission-controller/pull/78

Once the upstream bump is merge then we can a downstream merge sync which might be larger than this.

**Comment by miheer** [comment]
it looks like right way is to have a proper ds merge. i will update this PR to test the u/s to d/s merge.

**Comment by miheer** [comment]
@bpickard22 PTAL. 

u/s PR https://github.com/k8snetworkplumbingwg/net-attach-def-admission-controller/pull/78 is merged

**Comment by tssurya** [comment]
/retitle [CORENET-7375](https://redhat.atlassian.net/browse/CORENET-7375): d/s merge Bump Kubernetes to 1.36.2 and Go to 1.26

**Comment by tssurya** [comment]
as long as you title the PR correctly it all resolves. you had to add the `:`
